src/main/java/org/commcare/formplayer/session/FormplayerInstanceInitializer.java

```diff
@@ -61,10 +62,16 @@ protected InstanceRoot setupSessionData(ExternalDataInstance instance) {
 
         Hashtable<String, String> userProperties = u.getProperties();
 
+        String appLang = locale;
+        if (locale == null) {
+            String[] locales = Localization.getGlobalLocalizerAdvanced().getAvailableLocales();
+            appLang =  locales[locales.length-1];
```

Comment:
this grabs the last locale created? is that always right one?

Reply:
thanks so much for pointing this out, totally forgot to go back to this. The locales list is always as follows: ['default' (like literally just a string called default), <the actual language slug user set as default>, <all other slugs in order the were added]. The `locales.length-1` was previously a placeholder and wouldn't have worked if there were more than one languages. Added a similar comment in the code too to avoid confusion.